src/ucp/core/ucp_context.c

```diff
@@ -275,6 +298,21 @@ static ucs_status_t ucp_add_tl_resources(ucp_context_h context,
     return status;
 }
 
+static void ucp_check_unavailable_devices(str_names_array_t device_list, int *masks,
+                                     int index)
+{
+    int i;
+
+    if (masks[index] != -1) {
```

Comment:
this is slow path anyway, and the check is not needed and we can reduce code size by removing it